### PR TITLE
fetching call number types, locations, and libraries

### LINKED ISSUE #20: Dynamically pull and store reference data

### PR DESCRIPTION
**Pull request by ghost**
resolves #20

### CODE REVIEW

File: src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        private String name;
+        private Integer type;
+    }
+
+    private CallNumberType addType(CallNumberType callNumberType) {
+        String name = callNumberType.getName();
+        if (callNumberTypesMap.containsKey(name)) {
+            callNumberType.setType(callNumberTypesMap.get(name));
+        } else {
+            callNumberType.setType("0");
```
Comment (rladdusaw) [inline src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java]: I think the default should be 4 - other, Not LC

File: src/main/resources/map/callNumberTypes.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "Dewey": 1,
+  "Library of Congress": 0,
+  "National Library of Medicine": 0,
```
Comment (rladdusaw) [inline src/main/resources/map/callNumberTypes.json]: Though it will resolve to the same parsing, NLM should be type 2

**Review by rladdusaw — CHANGES_REQUESTED** [review]

File: src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java]: ok

File: src/main/resources/map/callNumberTypes.json (hunk 2 above)
Comment (ghost) [reply src/main/resources/map/callNumberTypes.json]: I will fix. I forgot to double check the values after creating.

File: src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/edu/tamu/modspineomatic/service/CallNumberTypeService.java]: Made it configurable with default to 4.

**Review by rladdusaw — APPROVED** [review]

**Pull request merged**